Fix number sensor string

Schouwenburg opened the pull request:
Hi, after fixing the number from string extraction I would also like to propose to prefer a float extraction above a int extraction. I ran into a use case where this really makes sense.

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [ ] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

murrant commented:
You are going to have troubles.  This is the quantastor string it is parsing `32 C (89.60 F)`  (LibreNMS wants temperatures in C)

Schouwenburg commented:
> You are going to have troubles. This is the quantastor string it is parsing `32 C (89.60 F)` (LibreNMS wants temperatures in C)

So i could skip this if sensor_class is temperature. What do you think, should I also skip this for sensor_class state?

PipoCanaja commented on tests/data/asa.json:
```diff
@@ -3131,126 +3131,6 @@
     "sensors": {
         "discovery": {
             "sensors": [
-                {
```
We see multiple sensors deleted. It probably shows that something is going wrong with the new code  preventing existing sensors to survive the change. You should end up with no sensor deleted if the new code is safe.

PipoCanaja commented on tests/data/bats.json:
```diff
@@ -202,7 +202,7 @@
                     "group": "Positioner",
                     "sensor_divisor": 1,
                     "sensor_multiplier": 1,
-                    "sensor_current": 17,
```
Here, the value looks very different, it must be checked if this is correct or not

murrant replied:
344.4 is incorrect, looking at the data (it should be 17).

PipoCanaja commented on tests/data/vrp_ar169sfp.json:
```diff
@@ -3509,102 +3509,6 @@
                     "sensor_prev": null,
                     "user_func": null,
                     "state_name": null
-                },
```
Same here. This device I have it running and would rather have a problem if temperature is broken :)

PipoCanaja reviewed (changes requested):
Hi @Schouwenburg 
Seems that the code is not completely accurate currently. I added a few comments but most of the JSON files with deleted code should not exist. Modified values are acceptable, but deleted sensors means loss of data.

Schouwenburg commented:
> Hi @Schouwenburg Seems that the code is not completely accurate currently. I added a few comments but most of the JSON files with deleted code should not exist. Modified values are acceptable (if they can be explained), but deleted sensors means loss of data.

Agree, I'm not yet sure where it goes wrong. I will do some debugging, should all be logical :-)

Jellyfrog commented:
First of all, thank you for your contribution!

We're closing this due to inactivity, feel free to reopen if you find the time to continue on this!
